spec/controllers/application_controller/buttons_spec.rb

```diff
@@ -340,13 +340,13 @@
 
     it "sets new role visibility for custom button" do
       controller.send(:button_set_record_vars, custom_button)
-      expect(custom_button.visibility[:roles]).to eq([role.name])
+      expect(custom_button.visibility[:roles]).to eq([role.id])
     end
 
     it "sets new role and preserves old role for custom button" do
       edit[:new][:roles] = [old_role.id, role.id.to_s] # old roles are represented by int and new ones by string
       controller.send(:button_set_record_vars, custom_button)
-      expect(custom_button.visibility[:roles]).to eq([old_role.name, role.name])
+      expect(custom_button.visibility[:roles]).to eq([old_role.id, role.id.to_s])
```

Comment:
TODO: that comment on line 347 is nuts.  I need to figure out if roles a strings and ints is an actual thing.